test/dotnet-help.Tests/GivenThatIWantToShowHelpForDotnetCommand.cs

```diff
@@ -65,5 +66,35 @@ public void WhenHelpOptionIsPassedToDotnetItPrintsUsage(string helpArg)
             cmd.Should().Pass();
             cmd.StdOut.Should().ContainVisuallySameFragment(HelpText);
         }
+
+        [Theory]
+        [InlineData("add", "Usage: dotnet add [arguments] [options] [command]")]
+        [InlineData("add package", "Usage: dotnet add <PROJECT> package [arguments] [options]")]
+        [InlineData("add reference", "Usage: dotnet add <PROJECT> reference [options] [args]")]
+        [InlineData("pack", "Usage: dotnet pack [arguments] [options] [args]")]
+        [InlineData("new", "Usage: dotnet new [arguments] [options]")]
+        [InlineData("restore", "Usage: dotnet restore [arguments] [options] [args]")]
+        [InlineData("build", "Usage: dotnet build [arguments] [options] [args]")]
+        [InlineData("publish", "Usage: dotnet publish [arguments] [options] [args]")]
+        [InlineData("run", "Usage: dotnet run [options] [[--] <additional arguments>...]]")]
+        [InlineData("test", "Usage: dotnet test [arguments] [options] [args]")]
+        [InlineData("migrate", "Usage: dotnet migrate [arguments] [options]")]
+        [InlineData("clean", "Usage: dotnet clean [arguments] [options] [args]")]
+        [InlineData("sln", "Usage: dotnet sln [arguments] [options] [command]")]
+        [InlineData("sln add", "Usage: dotnet sln <SLN_FILE> add [options] [args]")]
+        [InlineData("sln list", "Usage: dotnet sln <SLN_FILE> list [options]")]
+        [InlineData("sln remove", "Usage: dotnet sln <SLN_FILE> remove [options] [args]")]
+        [InlineData("nuget", "Usage: dotnet nuget [options] [command]")]
```

Comment:
I don't think that with the current implementation of `CommandLineApplication`, there's a clean way to ensure this. This is a common theme in the command line parsing bugs and behavioral inconsistencies I've seen. Each command under `dotnet` has too much control. This is also why we have no true unit tests over our command line parsing code. I tried starting with a unit test in this case but the behavior in question was all specified at the individual commands so in the interest of keeping the change small, I used an integration test.

The command line parsing work I've been doing for v2 is meant to address these things.